doc/man3/flux_future_set_init.adoc

```diff
@@ -0,0 +1,163 @@
+flux_future_set_init(3)
+=======================
+:doctype: manpage
+
+
+NAME
+----
+flux_future_set_init, flux_future_set_result, flux_future_set_error, flux_future_aux_get, flux_future_aux_set, flux_future_set_flux, flux_future_get_flux - support methods for future wrappers
+
+
+SYNOPSIS
+--------
+ #include <flux/core.h>
+
+ typedef void (*flux_future_init_f)(flux_future_t *f,
+                                    flux_reactor_t *r, void *arg);
+
+ void flux_future_set_init (flux_future_t *f,
+                            flux_future_init_f cb, void *arg);
+
+ void flux_future_set_result (flux_future_t *f,
+                              void *result, flux_free_f free_fn);
+
+ void flux_future_set_error (flux_future_t *f, int errnum);
+
+ void *flux_future_aux_get (flux_future_t *f, const char *name);
+
+ int flux_future_aux_set (flux_future_t *f, const char *name,
+                          void *aux, flux_free_f destroy);
+
+ void flux_future_set_flux (flux_future_t *f, flux_t *h);
+
+ flux_t *flux_future_get_flux (flux_future_t *f);
+
+
+DESCRIPTION
+-----------
+
+A Flux future represents some activity that may be completed with reactor
+watchers and/or message handlers.  It is intended to be wrapped
+by other classes that perform specific activities, providing a common idiom
+for synchronizing on the result.  Typically such a class provides
+class-specific wrappers for `flux_future_create()` and `flux_future_get()`,
+and implements a class-specific init callback.
+
+`flux_future_set_init()` registers an init callback.  The callback sets
+up class-specific watchers on the reactor to handle asynchronous events.
+The watchers must eventually call `flux_future_set_result()` or
+`flux_future_set_error()` to fulfill the future.  For more information
+about when the initialization callback is invoked, see REACTOR CONTEXTS
+below.
+
+`flux_future_set_result()` assigns an opaque _result_ value to the
+future, with optional destructor.  NULL is a valid result.
+The assignment fulfills the future and triggers synchronization,
+after which `flux_future_get()` can be used to obtain the _result_
+value without blocking.  The destructor is called when the future is
+destroyed.
+
+`flux_future_set_error()` similarly fulfills the future and triggers
+synchronization, but instead of assigning a result, it assigns an
+errno value, after which `flux_future_get()` returns -1 with errno
+set to this value.
+
+`flux_future_aux_set()` attaches an object _aux_, with optional destructor
```

Comment:
So this doesn't have anything to do with the promise/future part of this?  Its just extra data tacked on?

Does this really belong here, in that case?  All involved threads must already have the shared flux_future_t handle, so if the want other extraneous data shared they can easily share it in the same way, no?  Doesn't this distract from the futures code doing one thing well?